python/setup.py

```diff
@@ -601,6 +602,20 @@ def build(build_python, build_java, build_cpp):
             ]
             + runtime_env_agent_pip_packages
         )
+    # Vendor setproctitle which is a C extension by
+    # copying the so file to the ray/private/thirdparty/ folder.
+    subprocess.check_call(
```

Comment:
Actually why not just declare `setproctitle` as a default dependency of Ray? What's the drawback of it? https://discuss.python.org/t/can-vendoring-dependencies-in-a-build-be-officially-supported

Reply:
> Even if I don’t need a constrained version of it today, any time it does a breaking change (major release if you use semver), I need to consider it. jsonschema recently released 4.0, so using it would put us in the position of having to support jsonschema 3.x and 4.x in the same codebase.

well, this can be a reason for vendoring.

Reply:
Less vendoring is better in my opinion. `conda-forge` has been quite successful in patching ray to declare a regular dependency without vendoring.